views/topologicalSort.pug

```diff
@@ -1,9 +1,263 @@
+                        tan.push(t)
+                    }
+                }
+            }
+            tan.sort()
+            index = Math.floor(size/2 - 1) - right
+            mid_tan = tan[index]
+
+            // search path
+            var path = [mid_name]
+            mid_index = 0
+            for(name in vertex_dict){
+                key = vertex_dict[name]
+                key_x = key[0]
+                key_y = key[1]
+                if(key_x <= mid_x && key_y <= mid_y|| key_x*a + key_y*b < threLower) continue;
+                if(key_y - mid_y > 0 && (key_x - mid_x <= 0 || (key_y - mid_y)/(key_x - mid_x) > mid_tan || key_x*a + key_y*b > threUpper)){
+                    for(var i = 0; i < mid_index + 1; i++){
+                        var t_name = path[i]
+                        var t_point = vertex_dict[t_name]
+                        if (key_y > t_point[1]){
+                            mid_index++;
+                            path.splice(i, 0, name);
+                            break;
+                        }
+                    }
+                }
+                if(key_x - mid_x > 0 && (key_y - mid_y < 0 || (key_y - mid_y) / (key_x - mid_x) <= mid_tan || key_x*0.5 + key_y*0.5 > 6)){
+                    var t_name = path[path.length-1];
+                    var t_point = vertex_dict[t_name]
+                    if(key_x > t_point[0]){
+                        path.push(name);
+                    }else{
+                        for(var i = mid_index + 1; i < path.length; i++){
+                            if(key_x < vertex_dict[path[i]][0]){
+                                path.splice(i,0,name);
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            return path
+        }
+        
+        function drawPath(mid_name, path){
+            var line_stroke = "red"
+            var polyline_set = new Array()
+            for(i = 0; i<path.length-1;i++){
+                if(path[i] == mid_name){line_stroke = "green"};
+                var name1 = path[i]
+                var point1 = vertex_dict[name1]
+                var name2 = path[i+1]
+                var point2 = vertex_dict[name2]
+                var points = (50*point1[0] + 20) + "," + (520 - 50*point1[1]) + " " + (50*point2[0] + 20) + "," + (520 - 50*point2[1])
+                var style = "fill:none;stroke:" + line_stroke + ";stroke-width:2";  
+                var polyline = [points,style]
+                polyline_set.push(polyline)
+            }
+            svg.selectAll("polyline")
+                .data(polyline_set)
+                .enter()
+                .append("polyline")
+                .attr("points",function(d,i){
+                        return d[0]})
+                .attr("style", function(d,i){return d[1]})
+        }
+
+        function drawTable(mid_name, path){
```

Comment:
Thanks for remembering to add the table. Maybe this part can be moved into public/javascript so as to reuse it and show the result of genetic algorithm. Anyway, if you run out of time, I can do this part.

Reply:
Ok, I can do that.